tools/IceRpc.BuildTelemetry.Reporter/Program.cs

```diff
@@ -66,8 +65,25 @@
 
     BuildTelemetry buildTelemetry = idl.ToLower() switch
     {
-        "slice" => new BuildTelemetry.Slice(new SliceTelemetryData(version, compilationHash, containsSlice1, containsSlice2, sourceFileCount, referenceFileCount)),
-        "protobuf" => new BuildTelemetry.Protobuf(new ProtobufTelemetryData(version, compilationHash, sourceFileCount)),
```

Comment:
Reworked the code to use the generated constructor, and remove the other constructor that hardcodes the defaults